packages/cli/src/databases/repositories/execution.repository.ts

```diff
@@ -387,21 +387,47 @@ export class ExecutionRepository extends Repository<ExecutionEntity> {
 			customData,
 			...executionInformation
 		} = execution;
-		if (Object.keys(executionInformation).length > 0) {
-			await this.update({ id: executionId }, executionInformation);
-		}
 
-		if (data || workflowData) {
-			const executionData: Partial<ExecutionData> = {};
-			if (workflowData) {
-				executionData.workflowData = workflowData;
+		const { type: dbType, sqlite: sqliteConfig } = this.globalConfig.database;
+
+		if (dbType === 'sqlite' && sqliteConfig.poolSize === 0) {
+			// TODO: Delete this block of code once the sqlite legacy (non-pooling) driver is dropped.
+			// In the non-pooling sqlite driver we can't use transactions, because that creates nested transactions under highly concurrent loads, leading to errors in the database
+
+			if (Object.keys(executionInformation).length > 0) {
+				await this.update({ id: executionId }, executionInformation);
 			}
-			if (data) {
-				executionData.data = stringify(data);
+
+			if (data ?? workflowData) {
```

Comment:
nullish coalescing in an `if` statement looks our of place. can we please switch this back to `||` 🙏🏽